packages/compile-solidity/compilerSupplier/Cache.js

```diff
@@ -0,0 +1,37 @@
+const Config = require("@truffle/config");
+const path = require("path");
+const fs = require("fs");
+
+class Cache {
+  constructor() {
+    const compilersDir = path.resolve(
+      Config.getTruffleDataDirectory(),
+      "compilers"
+    );
+    const compilerCachePath = path.resolve(compilersDir, "node_modules"); // because babel binds to require & does weird things
```

Comment:
Didn't we just cache the compilers in the `compilers` directory? Do we put it in `node_modules` because that is ignored by babel processing when it comes to using `require` on a path in `node_modules`?

Reply:
I think it's for us... we have a package or two that uses babel, and IIRC it explodes without this.

Reply:
(this has been this way; I haven't changed this)

Reply:
oh I thought this was new, I think the issue is that babel processes all required items and the compiler is so big that it will explode trying to process it all

Reply:
oh cool, good to know!